src/python/gem5/resources/resource.py

```diff
@@ -554,6 +555,70 @@ def _get_weights_and_simpoints_from_file(
         return simpoint_list, weight_list
 
 
+class SuiteResource(AbstractResource):
+    def __init__(
+        self,
+        workloads: List[Dict[str, Any]],
+        local_path: Optional[str] = None,
+        resource_version: Optional[str] = None,
+        description: Optional[str] = None,
+        source: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            local_path=local_path,
+            description=description,
+            source=source,
+            resource_version=resource_version,
+        )
+        self.resource_version = resource_version
+        self.workloads = workloads
+        self.workload_obj_list = []
+        self.index = 0
```

Comment:
Maybe all these should be private?

```suggestion
        self._resource_version = resource_version
        self._workloads = workloads
        self._workload_obj_list = []
        self._index = 0
```